Sources/FINN/BackendModel/FilterSetup.swift

```diff
@@ -50,33 +50,35 @@ public struct FilterSetup: Decodable {
     public func filterContainer(using config: FilterConfiguration) -> FilterContainer {
         var rootSubfilters = [Filter]()
 
-        if let key = config.searchFilterKey {
+        if let key = config.searchFilter {
             rootSubfilters.append(Filter(title: "search_placeholder".localized(), key: key))
         }
 
-        if let key = config.preferencesFilterKey {
-            let preferenceSubfilters = config.preferenceFilterKeys.compactMap { filterData(forKey: $0) }
+        if let key = config.preferencesFilter {
+            let preferenceSubfilters = config.preferenceFilters.compactMap { filterData(forKey: $0) }
             let preferenceFilter = Filter(title: "", key: key)
             preferenceSubfilters.forEach { preferenceFilter.add(subfilter: $0.asFilter()) }
 
             rootSubfilters.append(preferenceFilter)
         }
 
-        let supportedFilters = config.supportedFiltersKeys.compactMap { key -> Filter? in
+        var supportedFilters = config.supportedFilters.compactMap { key -> Filter? in
             let kind: Filter.Kind = config.contextFilters.contains(key) ? .context : .normal
             return filterData(forKey: key)?.asFilter(of: kind)
         }
 
-        if let locationFilter = supportedFilters.first(where: { $0.key == config.mapFilterParentFilterKey }) {
+        if let key = config.mapFilter {
             let mapFilter = MapFilter(
                 title: "map_filter_title".localized(),
-                key: "map",
-                latitudeKey: "lat",
-                longitudeKey: "lon",
-                radiusKey: "radius",
-                locationKey: "geoLocationName"
+                key: key,
+                latitudeKey: FilterKey.latitude.rawValue,
+                longitudeKey: FilterKey.longitude.rawValue,
+                radiusKey: FilterKey.radius.rawValue,
+                locationKey: FilterKey.geoLocationName.rawValue
             )
-            locationFilter.add(subfilter: mapFilter, at: 0)
+
+            let index = supportedFilters.firstIndex(where: { $0.key == config.locationFilter }) ?? 0
```

Comment:
Could add map key into the `supportedFilters` at the correct index in the config.

Reply:
Great idea, now I know what's `supportedFilters` for 😄

Reply:
We could maybe come up with a better name for those filters :P

Reply:
`RootLevelFilters` or something is a bit more descriptive

Reply:
@shredlocker Should be fixed now 😉